spikeinterface/curation/auto_merge.py

```diff
@@ -0,0 +1,264 @@
+import numpy as np
+import scipy.signal
+import scipy.spatial
+
+from ..postprocessing import compute_correlograms, get_template_extremum_channel
+
+# TODO:
+#   * adaptative window p(aka plage) on CC
+#   * template similarity
+#   * 
+
+def get_potential_auto_merge(waveform_extractor,
+                minimum_spikes=1000, maximum_distance_um=200.,
+                peak_sign="neg",
+                bin_ms=0.25, window_ms=50., corr_thresh=0.3,
+                correlogram_low_pass = 800., adaptative_window_threshold=0.5,
+                debug_folder=None,
+                ):
+    """
+    Algorithm to find and check potential merges between units.
+    
+    This is taken from lussac version 1 done Aurelien Wyngaard.
+    https://github.com/BarbourLab/lussac/blob/main/postprocessing/merge_units.py
+    
+    This check:
+       * correlograms
+       * template_similarity
+       * contamination quality mertics
+    
+    """
+    
+    we = waveform_extractor
+    sorting = we.sorting
+    unit_ids = sorting.unit_ids
+    
+    # STEP 1 : to get fast computation we will not analyse pairs when:
+    #    * not enough spikes for one of theses
+    #    * to far away one from each other
+    n = unit_ids.size
+    pair_mask = np.ones((n, n), dtype='bool')
+    num_spikes = np.array(list(sorting.get_total_num_spikes().values()))
+    to_remove = num_spikes < minimum_spikes
```

Comment:
`minimum_spikes` isn't very precise, because 1,000 spikes in a 10s recording is not the same as 1,000 spikes in a 30m recording.
We want enough spikes so that the correlogram actually looks like something.

I think the threshold needs to be put on something like `n_spikes**2 / recording.get_num_samples()` (this roughly translate to the probability of 2 spikes meeting each other).

Reply:
Yep sure. I will find something